### DISCUSSION TITLE
Flag on parcel build to quickly export debug mode files to the filesystem

### DISCUSSION THREAD

**Discussion by john01dav** [parcel-bundler/parcel#5381] [Q&A] [answered]
# 🙋 feature request

In my use case for Parcel, I need to integrate it with my backend. The way that that works is to have my backend's build system (Cargo) scan Parcel's output directory (dist) and then link all of those files into my resulting executable binary to be served (via [this library](https://crates.io/crates/actix-web-static-files)). Right now, the way that debug builds work is that my IDE first calls Parcel to export a release build, and then builds my backend in debug mode, incorporating Parcel's release files. It would be better if, instead, I could configure it to tell Parcel to put out debug-mode files on the filesystem, so that when building in debug those can be incorporated. It seems like a debug build is faster, and the non-minified JS can be easier to debug. I mostly care about the speed. The normal integrated web server for debug mode, while generally useful, isn't useful in my use case: my code is designed to all be served from the same web server.

## 🤔 Expected Behavior
Ideally there could be a flag to build, so, for example, `npx parcel build --debug` would export in debug mode. Note that I'm not asking for any sort of file watching with this, as that isn't compatible with my backend system anyway.

## 😯 Current Behavior
No such flag exists that I'm aware of.

## 💁 Possible Solution
I'm not aware of any current workarounds to this (although if the reader is, please share)

## 🔦 Context
The main reason for this feature request is to save time waiting on release builds constantly.

## 💻 Examples
I do not have any examples, but I think that what I'm saying is clear without them.

**Comment by DeMoorJasper** [comment, marked as answer]
What do you mean with debug mode? There is no debug mode in Parcel.

Do you mean a development build (a non production optimised build)?

If you mean that, this already exists... it's just `NODE_ENV=development parcel build...`

**Reply by john01dav** [reply]
Yep, that's what I mean. Thank you :)
